tests/integration/actions/settings/base.py

```diff
@@ -0,0 +1,133 @@
+"""Base class for ``settings`` interactive/stdout tests."""
+
+import difflib
+import os
+
+import pytest
+
+from ....defaults import FIXTURES_DIR
+from ..._common import retrieve_fixture_for_step
+from ..._common import update_fixtures
+from ..._interactions import SearchFor
+from ..._interactions import UiTestStep
+from ..._tmux_session import TmuxSession
+
+
+TEST_FIXTURE_DIR = os.path.join(FIXTURES_DIR, "integration/actions/settings")
+
+base_steps = (
+    UiTestStep(user_input=":f app", comment="filter for app settings"),
+    UiTestStep(user_input=":0", comment="app settings details"),
+    UiTestStep(user_input=":back", comment="return to filtered settings list"),
+    UiTestStep(
+        user_input=":f",
+        comment="clear filter, full list",
+        present=["ansible_runner_artifact_dir", "help_playbook"],
+        mask=True,
+    ),
+    UiTestStep(user_input=":f exec", comment="filter using a different index"),
+    UiTestStep(user_input=":3", comment="execution_environment_image details"),
+    UiTestStep(user_input=":back", comment="return to filtered list"),
+    UiTestStep(
+        user_input=":f",
+        comment="clear filter, full list",
+        present=["ansible_runner_artifact_dir", "help_playbook"],
+        mask=True,
+    ),
+)
+
+
+class BaseClass:
+    """Base class for interactive/stdout ``settings`` tests."""
+
+    UPDATE_FIXTURES = False
+    PANE_HEIGHT = 25
+    PANE_WIDTH = 300
+
+    @pytest.fixture(scope="module", name="tmux_session")
+    def fixture_tmux_session(self, request):
+        """Tmux fixture for this module.
+
+        :param request: Used for generating test id
+        :yield: tmux_session object
```

Comment:
I think it's
```suggestion
        :yields: tmux_session object
```

Reply:
I don't think sphinx cares does it? https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#python-signatures

Both will pass the darglint linter being vetted.  That said, I have been trying to use `:returns:` vs `:return:` so `:yields:` would follow that pattern and is how it is done elsewhere in navigator

Reply:
Interesting. I was sure the darglint and Sphinx expect plural versions since this is what I've seen usually.
I would argue that the phrase would be expected to be a continuation of the keyword, as in "returns ...some..stuff" vs a call-to-action "return ...some..things". Keep in mind, that Sphinx will render this as plural `Returns: <explanation>` in HTML so that may be an additional reason to stick with the plural version.

Reply:
changed to plural, also think sticking to a particular pattern with plurals is a good idea.